src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -79,17 +79,23 @@ public List<HighPerformanceDetection> Scan(StringInput input)
             return [];
         }
 
+#if NET
+        HighPerformanceStringInput highPerformanceInput = input.Span;
```

Comment:
It's tempting to use StringInput throughout in this file, but that leads to really chatty Memory -> Span conversion because we do things like look at a single character in the input. Probably worth a comment. 

If we could move everything to Span from Memory, this is another thing we wouldn't have to worry about.